gusto/time_discretisation/explicit_runge_kutta.py

```diff
@@ -267,6 +397,67 @@ def solve_stage(self, x0, stage):
                 if self.limiter is not None:
                     self.limiter.apply(self.x1)
 
+        elif self.rk_formulation == RungeKuttaFormulation.linear:
+
+            # Set combined index of stage and subcycle
+            cycle_stage = self.nStages*self.subcycle_idx + stage
+
+            if stage == 0 and self.subcycle_idx == 0:
+                self.field_lhs = [Function(self.fs) for _ in range(self.nStages*self.ncycles)]
+                self.field_lhs[0].assign(self.x0)
+
+            # All-but-last form ------------------------------------------------
+            if (cycle_stage + 1 < self.ncycles*self.nStages):
+                # Build up RHS field to be evaluated
+                self.field_rhs.assign(0.0)
+                for i in range(stage+1):
+                    i_cycle_stage = self.nStages*self.subcycle_idx + i
+                    self.field_rhs.assign(
+                        self.field_rhs
+                        + self.butcher_matrix[stage, i]*self.field_lhs[i_cycle_stage]
+                    )
+
+                # Evaluate physics and apply limiter, if necessary
+                for evaluate in self.evaluate_source:
+                    evaluate(self.field_rhs, self.dt)
+                if self.limiter is not None:
+                    self.limiter.apply(self.field_rhs)
```

Comment:
Do we want to apply the limiter here, or after the solve?

Reply:
It should be here, as it is applied on the right-hand side field that is evaluated. There is a final application of the limiter at the end after the final solve